app/services/reporter.rb

```diff
@@ -6,61 +6,80 @@
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
 class Reporter
-  attr_reader :storage_root, :druids
+  attr_reader :storage_root
 
   # @params [Hash] params used to initialize the Reporter service
   # @return [Reporter] the reporter
   def initialize(params)
     @storage_root = MoabStorageRoot.find_by!(name: params[:storage_root_name])
-    @druids = moab_storage_root_druid_list
+    @msr_names = {}
   end
 
-  # @return [Array] an array of druids on the storage root
-  def moab_storage_root_druid_list
-    druid_array = []
+  def moab_storage_root_list_pres_obj_comp_moabs_relation
```

Comment:
I appreciate renaming this method to be more descriptive of what it's doing. If I'm grokking (and I may not be), this method is primarily operating on PreservedObjects, using CompleteMoabs only to filter the POs in the given storage root. If that is true, I wonder about a tweak to the name like `preserved_objects_in_storage_root`? (The name currently is almost Germanic in its love for compound nouns, which is impressive. :smile:)

Reply:
And I might also like to see some YARD documentation to show that it returns an AR relation. (Would rather see that in YARD than in the method name, but I'm open to being overruled there! For me, I think having `preserved_objects` in the method name already implies it's an AR relation but maybe that's just me.)

Reply:
> The name currently is almost Germanic in its love for compound nouns, which is impressive

ha, glad you like it, i think my tendency for that drives @ndushay crazy 😆 

> this method is primarily operating on PreservedObjects, using CompleteMoabs only to filter the POs in the given storage root

sorta.  i found it more useful to think of this in SQL than in ActiveRecord objects.

this method is really about getting all the `CompleteMoabs` on a storage root, and all their relevant details.  `PreservedObject` is the master record for an object in the abstract, tying together its uncompressed moabs on storage roots and the zip copies in the cloud (and the preservation policies that say what goes where).  as such, the `druid` lives on the `PreservedObject`.  that's the main point of tying in the `PreservedObject`.

so really i want all the info for `CompleteMoab`s on the storage root, plus some info that lives on the related `PreservedObject` and `MoabStorageRoot` rows.  might've been better to have the query be rooted in `PreservedObject`, joining to `CompleteMoab`?  i don't have any strong feelings on that, esp since all i really want to do is get the query results as a list of column values, without bothering to instantiate ActiveRecord objects.

i think this is just a natural artifact of the tendency for OOP to clash with relational modeling at the seams where the two meet.  since this is a query for a large number of things that we want to run correctly and efficiently, i erred on the side of writing good SQL (in the most maintainable rails way i could imagine) rather than erring on the side of writing things in the rails-iest way.  but i also wanted to use AR relation composability instead of just writing straight SQL or something.  very open to feedback on how to check all those boxes in a cleaner way.

Reply:
> And I might also like to see some YARD documentation to show that it returns an AR relation. (Would rather see that in YARD than in the method name, but I'm open to being overruled there! For me, I think having preserved_objects in the method name already implies it's an AR relation but maybe that's just me.)

works for me.  will try to get to a terser name in light of YARD doc for return type.  if you have suggestions on naming in light of the data i'm trying to get back out of the relation, i'm all ears.

Reply:
Aha! Now that I look at how this relation is used and cross-reference it with the database schema, I **now** see that the PreservedObject is primarily for the druid, and that all of the other fields come from the CompleteMoab. How about a name like `complete_moabs_and_preserved_objects_in_storage_root` or some such? This is me saying that I am OK with verbose method names and that I also prefer in-domain language over abbreviations, which is :100: subjective.

Reply:
I'm thinking this method should be private.  Does that ring true for you?

Reply:
@ndushay I think that makes good sense.

Reply:
@ndushay adding to that, I think only these methods should be public:

* `#initialize`
* `#write_to_csv`
* `#druid_csv_list`
* `#moab_detail_csv_list`

The rest could/should be private to constrain the interface of the reporter and hide its implementation details.

Reply:
> How about a name like `complete_moabs_and_preserved_objects_in_storage_root` or some such?

i like it, will go w/ that

Reply:
i took all the suggestions for privatization, except for `#default_filepath`:  it was easier to leave that public in this follow up because it kept me from having to change the tests at all, and it seems defensible to leave it in the public interface (but that's me rationalizing convenience, and i'm happy to privatize that too, but would prefer to do so in a follow-on PR).

Reply:
I'm not seeing the changes yet, @jmartin-sul. did you push?

Reply:
`#default_filepath` will be made private as part of addressing #1367